### PR TITLE
Copy editing the Release Notes role handbook

### PR DESCRIPTION
**Pull request by marpaia**
This PR contains a few copy-edits and markdown formatting changes to the Release Notes Handbook.

### CODE REVIEW

File: release-team/role-handbooks/release-notes/README.md [hunk 1]
```diff
@@ -2,32 +2,21 @@
...
-use a reasonable amount of creative liberity in completing the notes
+- The Release notes lead and shadows attend burn down meetings, SIG Release meetings and follow the [#sig-release](https://kubernetes.slack.com/messages/C2C40FMNF) Slack channel for relevant information throughout the release cycle.
+- One member of the release notes team should be responsible for setting up and running the [release notes tool](https://github.com/marpaia/release-notes) to collect generated drafts of all release-notes identified in the current release -- improvements to the tool are always welcome!
+- Drafts produced by the tool are copied to a Google Document that gives editing privleges to each member of the release-notes team (there is currently no automated way to keep the generated release-notes and the google doc concurrent. Some tool to automate this process would be a good task for the 1.13 release team)
+- Closer to the release date, major themes are collected from the release notes draft and the release notes team creates a draft in markdown of the identified major themes to send to sig-leads for feedback/confirmation
```
Comment (AishSundar) [inline release-team/role-handbooks/release-notes/README.md]: Is it safe to say we want to send the initial draft and release themes to SIG leads for review during Code slush or immediately after Code freeze begins.

**Comment by marpaia** [comment]
Great point @AishSundar, I updated the steps there to be more explicit about when we should start reaching out to SIG leads.

**Comment by AishSundar** [comment]
/lgtm

Thanks for updating this. If you are making change to the release notes sourcing process i.e., looking at k/enhancement repo issues for the milestone, weekly LWKD report etc, it will be good to update this handbook with those sources as well. May be a followup PR?

**Comment by AishSundar** [comment]
Holding this for others to review as well. Feel free to remove hold as appropriate.

**Pull request merged**